reanalysis/hail_filter_and_classify.py

```diff
@@ -0,0 +1,676 @@
+                        lambda x: hl.len(
+                            critical_consequences.intersection(
+                                hl.set(x.consequence_terms)
+                            )
+                        )
+                        > 0
+                    )
+                )
+                & (
+                    (
+                        matrix.vep.transcript_consequences.any(
+                            lambda x: (x.lof == LOFTEE_HC) | (hl.is_missing(x.lof))
+                        )
+                    )
+                    | (matrix.info.clinvar_sig.lower().contains(PATHOGENIC))
+                ),
+                ONE_INT,
+                MISSING_INT,
+            )
+        )
+    )
+
+
+def annotate_class_4(matrix: hl.MatrixTable, config: Dict[str, Any]) -> hl.MatrixTable:
+    """
+    Class based on in silico annotations
+    - rare in Gnomad, and
+    - CADD & REVEL above threshold (switched to consensus), or
+    - Massive cross-tool consensus
+    - polyphen and sift are evaluated per-consequence
+    :param matrix:
+    :param config:
+    :return:
+    """
+
+    return matrix.annotate_rows(
+        info=matrix.info.annotate(
+            Class4=hl.if_else(
+                (
+                    (matrix.info.cadd > config.get('cadd'))
+                    & (matrix.info.revel > config.get('revel'))
+                )
+                | (
+                    (
+                        matrix.vep.transcript_consequences.any(
+                            lambda x: hl.or_else(x.sift_score, MISSING_FLOAT_HI)
+                            < config.get('sift')
+                        )
+                    )
+                    & (
+                        matrix.vep.transcript_consequences.any(
+                            lambda x: hl.or_else(x.polyphen_score, MISSING_FLOAT_LO)
+                            > config.get('polyphen')
+                        )
+                    )
+                    & (
+                        (matrix.info.mutationtaster.contains('D'))
+                        | (matrix.info.mutationtaster == 'missing')
+                    )
+                    & (matrix.info.gerp_rs >= config.get('gerp'))
+                    & (matrix.info.eigen_phred > config.get('eigen'))
+                ),
+                ONE_INT,
+                MISSING_INT,
+            )
+        )
+    )
+
+
+def filter_matrix_by_ac(
+    matrix_data: hl.MatrixTable, config: Dict[str, Any]
+) -> hl.MatrixTable:
+    """
+
+    :param matrix_data:
+    :param config:
+    :return:
+    """
+
+    # count the samples in the VCF, and use to decide whether to implement
+    # 'common within this joint call' as a filter
+    # if we reach the sample threshold, filter on AC
+    if matrix_data.count_cols() >= config['min_samples_to_ac_filter']:
+        matrix_data = matrix_data.filter_rows(
+            matrix_data.info.AC <= matrix_data.info.AN // config['ac_filter_percentage']
```

Comment:
Did you mean to use integer division (`//`) here?

Reply:
the AC is always an int, so it's a case of float division and `<`, or int division and `<=`. This AC filter at the moment is entirely arbitrary, so I'll raise it in the progress meeting to see if there are strong feelings on if/how to implement this

Reply:
Cool, thanks! It's very likely that I misunderstand the intent of the filtering here, but naïvely I would have thought that this would be an expression of the form `AC / AN > threshold`. But that would translate to `AC > AN * threshold` or alternatively `AN < AC / T`, but you have `AC < AN / T` here.

Reply:
Yeah, I think this is one to revisit in the config definition. For some reason in the config I put an int, so the calculation is `AC < AN / 8` which is just stupid. 

Will switch this back to a proportion and make this a multiple

`AN/AC > threshold`